api/naming_service.go

```diff
@@ -0,0 +1,110 @@
+package api
+
+import (
+	"net/http"
+	"strconv"
+	"strings"
+
+	"encoding/hex"
+
+	"github.com/gin-gonic/gin"
+	"github.com/spf13/viper"
+	CoinType "github.com/trustwallet/blockatlas/coin"
+	"github.com/trustwallet/blockatlas/pkg/blockatlas"
+
+	"github.com/ethereum/go-ethereum/ethclient"
+	ens "github.com/wealdtech/go-ens/v3"
+)
+
+type ZNSResponse struct {
+	Addresses map[string]string
+	Meta      struct {
```

Comment:
can we avoid nested structs? 

did you consider splitting logic into separate files? for Zil, ETH and FIO in the future?

Reply:
good suggestion, will try to split it into different platforms

Reply:
done